java/java-impl/src/com/intellij/codeInsight/completion/JavaGenerateMemberCompletionContributor.java

```diff
@@ -45,38 +46,84 @@ public static void fillCompletionVariants(CompletionParameters parameters, Compl
     PsiElement position = parameters.getPosition();
     if (psiElement(PsiIdentifier.class).withParents(PsiJavaCodeReferenceElement.class, PsiTypeElement.class, PsiClass.class).
       andNot(JavaKeywordCompletion.AFTER_DOT).accepts(position)) {
+      int priority = -1;
+      boolean followedByWhitespace = psiElement()
```

Comment:
I suspect that written as plain old `if`, that might be more debuggable.

Reply:
You mean instead of using Psi pattern matching, write out the checks using "manual" Psi traversals and checks?

Reply:
Yes. Some `PsiTreeUtil` methods might be handy here, or maybe a `JBIterable.generate(e, PsiElement::getPrevSibling)`